.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,27 @@
+Thank you for contributing to JanusGraph.
+
+In order to streamline the review of the contribution we ask you
+to ensure the following steps have been taken:
+
+### For all changes:
+- [ ] Is there an issue associated with this PR? Is it referenced 
+     in the commit message?
+
+- [ ] Does your PR title ends with #xyz where xyz is the issue number you are trying to resolve?
```

Comment:
Instead of putting it in the title, please put the issue # in the body. I've edited a few issues that do the former to do the latter, and here's why:

* if the issue # is in the title, it looks confusing, because GitHub also appends the issue # to the title, so it looks like `this PR addresses issue #A #B` where #A is the issue you're addressing and #B is the auto-assigned number for this PR.

* the `#A` in the title is not auto-linked, so you can't click on it to visit the issue

* the issue `#A` does not automatically get a reference to the fact that PR #B is going to address it

In contrast, if you put `addresses issue #A` into the PR body, all of the above caveats are reversed.